mirage/awa_mirage.ml

```diff
@@ -27,7 +30,9 @@ module Make (F : Mirage_flow.S) (T : Mirage_time.S) (M : Mirage_clock.MCLOCK) =
   let write_flow t buf =
     FLOW.write t.flow buf >>= function
     | Ok () -> Lwt.return (Ok ())
-    | Error w -> t.state <- `Error (`Write w) ; Lwt.return (Error (`Write w))
+    | Error w ->
+      Log.warn (fun m -> m "error %a while writing" F.pp_write_error w);
+      t.state <- `Error (`Write w) ; Lwt.return (Error (`Write w))
```

Comment:
I'm not familiar with this flow, but I could worry we report the same error multiple places since we both log it and return it upstream.

Reply:
eventually this will indeed be reported. My thought was to report the error as early as possible, and I am not too worried about duplicated log messages.